presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java

```diff
@@ -144,42 +148,49 @@ public void configure(ResourceGroup group, SelectionContext context)
         return (!globalProperties.isEmpty()) ? globalProperties.get(0).getCpuQuotaPeriod() : Optional.empty();
     }
 
-    @VisibleForTesting
-    public synchronized void load()
+    private synchronized void refresh()
     {
         try {
-            Map.Entry<ManagerSpec, Map<ResourceGroupIdTemplate, ResourceGroupSpec>> specsFromDb = buildSpecsFromDb();
-            ManagerSpec managerSpec = specsFromDb.getKey();
-            Map<ResourceGroupIdTemplate, ResourceGroupSpec> resourceGroupSpecs = specsFromDb.getValue();
-            Set<ResourceGroupIdTemplate> changedSpecs = new HashSet<>();
-            Set<ResourceGroupIdTemplate> deletedSpecs = Sets.difference(this.resourceGroupSpecs.keySet(), resourceGroupSpecs.keySet());
+            load();
+        }
+        catch (Throwable e) {
```

Comment:
I guess this change of wrapping `load` with `try/catch` is not related to the checks you added in the constructor, but for making sure the scheduler thread doesn't die. If this is correct, it will make sense to move this change to a separate commit.